### DISCUSSION TITLE
How do I suppress listing directories on a new prompt?

### DISCUSSION THREAD

**Discussion by ghost** [marlonrichert/zsh-autocomplete#259] [Q&A] [answered]
First of all, excellent job for the very useful tool provided. I installed it an I am running it on fedora 34. The current directory is stuck in in "ls" mode. There us any one of the modules i can remove?

╭─      ~ ····························································································· ✔  at 22:10:56   ─╮
╰─                                                                                                                                ─╯
directory
Android/  Desktop/  Documents/  Downloads/  eclipse-workspace/  libfprint/  Music/  Pictures/  Public/  snap/  Templates/  Videos/

**Comment by marlonrichert** [comment, marked as answer]
What do you mean with "stuck"? Can you make me a reproducible test case?

---

Actual answer, [from the discussion below](https://github.com/marlonrichert/zsh-autocomplete/discussions/259#discussioncomment-1404629):

```zsh
zstyle ':autocomplete:*' min-input 1
```

**Reply by jasonvarga** [reply]
![image](https://user-images.githubusercontent.com/105211/135021021-82e8a4bd-4cfa-499b-9cb3-9385e36b7cb2.png)

I believe I'm running into the same issue.

On a blank prompt, the autocomplete is showing the directory listing.

**Reply by jasonvarga** [reply]
Adding `unsetopt autocd` to my zshrc "fixes" it, but I lose the auto cd feature.
https://github.com/marlonrichert/zsh-autocomplete/discussions/308#discussioncomment-1025233

**Reply by marlonrichert** [reply]
> On a blank prompt, the autocomplete is showing the directory listing.

Yes, that’s normal. But are you now “stuck”? If so, in what way?

**Reply by marlonrichert** [reply]
> Adding `unsetopt autocd` to my zshrc "fixes" it

Fixes what? What is broken?

**Reply by jasonvarga** [reply]
I don't think anything is really broken.
I think it's just unexpected to some people to see the directories when you haven't typed anything, that's all.

For example, if you're going to suggest directories, why just directories? Why not suggest every other possible command there too? You could type anything at a blank prompt.

And by "stuck", I think he means that it shows on screen. For me at least, nothing is actually stuck or frozen.

Anyway, I'm not complaining! Personally I prefer seeing an empty prompt until I start typing. I went with your other suggestion and overrode the function.

```sh
_autocomplete.config.tag-order() {
    local -aU tags=()
    [[ -n "$path[(r).]" ]] &&
        tags+=( globbed-files executables directories )
    [[ -o autocd ]] &&
        # tags+=( '(|local-)directories' )
    [[ -n $tags ]] &&
        tags+=( suffix-aliases )
    reply=( "$tags" ${${:-$PREFIX$SUFFIX}:--} )
}
```

**Reply by marlonrichert** [reply]
> if you're going to suggest directories, why just directories? Why not suggest every other possible command there too? You could type anything at a blank prompt.

Because then you can quickly see what you can do in the directory that you're in. 🙂  Also, it lets you press <kbd>↓</kbd> to drill down into subdirectories.

> And by "stuck", I think he means that it shows on screen. For me at least, nothing is actually stuck or frozen.

Ah, OK. I misunderstood that.

> I went with your other suggestion and overrode the function.

Actually, there's an easier solution! 😀
```zsh
zstyle ':autocomplete:*' min-input 1
```

**Reply by marlonrichert** [reply]
@cyber-arknet Does this solve your problem?

**Reply by jasonvarga** [reply]
`min-input 1` is much simpler. 🤣  Thanks!
